### PR TITLE
training: close resources

### PR DESCRIPTION
**Pull request by jburel**
# What this PR does

Make sure resources are closed

# Testing this PR

check that the examples are green

# Related reading
https://github.com/openmicroscopy/openmicroscopy/pull/5125

cc @bramalingam

### CODE REVIEW

File: examples/Training/java/src/training/RawDataAccess.java [hunk 1]
```diff
@@ -141,7 +141,9 @@ private void retrieveTile()
             }
         } catch (Exception e) {
             throw new Exception("Cannot read the tiles", e);
-        } 
+        } finally {
+            rdf.close();
```
Comment (joshmoore) [inline examples/Training/java/src/training/RawDataAccess.java]: Does this suggest eventually making the `Facility` interfaces subclass `Closeable`?

Comment (mtbc) [reply examples/Training/java/src/training/RawDataAccess.java]: Relevant to https://trello.com/c/K4d0rFQc/712-try-with-resources-for-blitz-services perhaps.

Comment (jburel) [reply examples/Training/java/src/training/RawDataAccess.java]: @joshmoore @mtbc certainly something to look into
so we can simplify good portion of code

**Comment by dominikl** [comment]
There's actually no `close` method in the `Facility` classes, the services are closed on log out, when the `Connector`s are closed: https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/blitz/src/omero/gateway/Connector.java#L857

**Comment by jburel** [comment]
That's something we may want to add to close services earlier e.g. rawfilestore
so we could then take advantages of ``try with resources``

**Comment by jburel** [comment]
card https://trello.com/c/Uavp5lOO/30-gateway-facility

**Comment by dominikl** [comment]
Looks good 👍

**Pull request merged**